src/main/java/com/netflix/conductor/kafka/eventqueue/KafkaObservableQueue.java

```diff
@@ -0,0 +1,408 @@
+             * to get the partition information.
+             */
+            this.consumer = new KafkaConsumer<>(consumerProperties);
+            this.consumer.subscribe(this.topics);
+        } catch (KafkaException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void start() {
+        logger.info("Started listening to {}:{}", getClass().getSimpleName(), queueName);
+        running = true;
+    }
+
+    @Override
+    public void stop() {
+        logger.info("Stopped listening to {}:{}", getClass().getSimpleName(), queueName);
+        running = false;
+    }
+
+    @Override
+    public boolean isRunning() {
+        return running;
+    }
+
+    /**
+     * Checks mandatory configs are available for kafka consumer.
+     *
+     * @param consumerProps
+     */
+    private void checkConsumerProps(Properties consumerProps) {
+        List<String> mandatoryKeys = Arrays.asList(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG);
+        List<String> keysNotFound = hasKeyAndValue(consumerProps, mandatoryKeys);
+        if (keysNotFound.size() > 0) {
+            String error = String.format("Configuration missing for Kafka consumer. %s", keysNotFound);
+            logger.error(error);
+            throw new RuntimeException(error);
+        }
+    }
+
+    /**
+     * Validates whether the property has given keys.
+     *
+     * @param prop
+     * @param keys
+     * @return
+     */
+    private List<String> hasKeyAndValue(Properties prop, List<String> keys) {
+        List<String> keysNotFound = new ArrayList<>();
+        for (String key : keys) {
+            if (!prop.containsKey(key) || Objects.isNull(prop.get(key))) {
+                keysNotFound.add(key);
+            }
+        }
+        return keysNotFound;
+
+    }
+
+    @Override
+    public Observable<Message> observe() {
+        OnSubscribe<Message> subscriber = getOnSubscribe();
+        return Observable.create(subscriber);
+    }
+
+    @Override
+    public List<String> ack(List<Message> messages) {
+        commitAsyncOffsets();
+        List<String> messageIds = new ArrayList<>();
+        for (Message message : messages) {
+            String[] idParts = message.getId().split(":");
+            int partitionNumber = Integer.parseInt(idParts[2]);
+            for (String topic : this.topics) {
+                for (PartitionInfo partition : consumer.partitionsFor(topic)) {
+                    if (partitionNumber == partition.partition()) {
+                        Map<TopicPartition, OffsetAndMetadata> currentOffsets = new HashMap<>();
+                        currentOffsets.put(new TopicPartition(idParts[1], partitionNumber),
+                                new OffsetAndMetadata(Integer.parseInt(idParts[3]) + 1, "no metadata"));
+                        try {
+                            consumer.commitSync(currentOffsets);
```

Comment:
Should we call commitsync for all messages from batch in a single call?

Reply:
The contract of the method involves the use of selective commit to return IDs of failed commits, but maybe it make sense to rewrite it